source/slang/slang-compiler.cpp

```diff
@@ -1550,20 +1628,31 @@ SlangResult dissassembleDXILUsingDXC(
 
         switch (target)
         {
+        case CodeGenTarget::HostCallable:
         case CodeGenTarget::SharedLibrary:
         case CodeGenTarget::Executable:
             {
+                ComPtr<ISlangSharedLibrary> sharedLibrary;
+
                 List<uint8_t> code;
                 if (SLANG_SUCCEEDED(emitCPUBinaryForEntryPoint(
                     compileRequest,
                     entryPoint,
                     entryPointIndex,
                     targetReq,
                     endToEndReq,
+                    sharedLibrary,
                     code)))
                 {
-                    maybeDumpIntermediate(compileRequest, code.getBuffer(), code.getCount(), target);
-                    result = CompileResult(code);
+                    if (target == CodeGenTarget::HostCallable)
```

Comment:
The amount of code overlap here between the `HostCallable` case and the other two is such that I don't know if lumping them together is worth it. If `emitCPUBinaryForEntryPoint` gets split along the lines I proposed above I'd definitely say these should be distinct cases.